functions/New-DbaDatabase.ps1

```diff
@@ -0,0 +1,469 @@
+        if ($UserDataFileGrowth -gt $UserDataFileGrowth)
+        {
+            Stop-Function -Message "UserDataFileGrowth of $UserDataFileGrowth is greater than the UserFileMaxSize setting of $UserDataFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+        #Check that the primary data file max size is greater than the primary data file size
+        if ($PrimaryFileSize -gt $PrimaryFileMaxSize)
+        {
+            Stop-Function -Message "PrimaryFilesize of $PrimaryFileSize is greater than the PrimaryFileMaxSize setting of $PrimaryFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+        #Check that the user data file max size is greater than the user file growth
+        if ($PrimaryFileGrowth -gt $PrimaryFileGrowth)
+        {
+            Stop-Function -Message "PrimaryFileGrowth of $PrimaryFileGrowth is greater than the PrimaryFileMaxSize setting of $PrimaryFileMaxSize" -Category InvalidData `
+            -ErrorRecord $_
+        }
+
+
+    }
+
+
+    process 
+    {
+        
+        if (Test-FunctionInterrupt) { return }
+
+        foreach ($Instance in $SqlInstance) 
+        {
+
+            #Verbose message to show the server
+            Write-Message -message "Connecting to server $Instance" -Level Verbose
+
+            #instantiate the sql server object
+	        try {
+		        $SQLServer = Connect-SqlInstance -SqlInstance $Instance -SqlCredential $SqlCredential
+	        }
+	        catch {
+		        Stop-Function -Message "Failure connecting to $Instance" -Category ConnectionError -ErrorRecord $_ -Target $Instance -Continue
+	        }
+
+            #check to see if the database already exists.
+            if ($SQLServer.Databases[$DatabaseName].Name -ne $Null)
+            {
+                Stop-Function -Message "Database $DatabaseName already exists on $Instance" -Target $Instance -Continue
+            }
+
+            # if we are using the default file locations, get them from the server
+            if ($UseDefaultFileLocations -eq $true)
+            {
+                #get the default file locations; if the master is in the default location we use that path as the default file will not be set
+                $LocalDataDrive = if ($SQLServer.DefaultFile -eq [DBNULL]::value)
+                                            {$SQLServer.MasterDBPath} 
+                                        else
+                                            {$SQLServer.DefaultFile}
+                $LocalLogDrive = $SQLServer.DefaultLog
+            }
+            elseif ($UseDefaultFileLocations -eq $false)
+            {
+                $LocalDataDrive = $NonDefaultFileLocation
+                $LocanewlLogDrive = $NonDefaultLogLocation
+            }
+
+            #create the file locations if they do not already exist
+            try
+            {
+                if ((test-path -path $LocalDataDrive) -eq $false)
```

Comment:
Good point, this was a recent addition that I've only tested locally. Will have a look at the best options for this.